### PR TITLE
Add release instructions and a helpful script

### PR DESCRIPTION
**Pull request by jskeet**
(Not sure how much of the automation actually happens yet, but we can edit that after trying...)

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Releasing the .NET Functions Frameworks packages
+
+This is a largely automated process. Steps:
+
+- Decide on a new version number, following semantic versioning, but
+  don't change any files.
+- Document the new version in [docs/history.md]() (but don't commit)
+- Run `./create-release-commit.sh {new-version}` from the root directory, passing
```
Comment (amanda-tarafa) [inline RELEASING.md:8]: nit: incomplete, "passing..."

File: create-release-commit.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+fi
+
+# Update CommonProperties.xml
+sed -i -e "s/<Version>.*<\/Version>/<Version>$NEW_VERSION<\/Version>/g" src/CommonProperties.xml
+
+# Update all references/templates
+./update-project-references.sh
+
+# Update the README instructions for template installation.
+# The binary mode )-b) here is to preserve existing line endings.
```
Comment (amanda-tarafa) [inline create-release-commit.sh:34]:
```suggestion
# The binary mode (-b) here is to preserve existing line endings.
```

File: create-release-commit.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+# (At some point we should probably make all line endings consistent...)
+sed -i -b -e "s/Templates::$OLD_VERSION/Templates::$NEW_VERSION/g" README.md
+
+# Build the commit message up as a file for simplicity
+mkdir -p tmp
+
+echo "Release Functions Framework .NET packages version $NEW_VERSION" > tmp/commit.txt
+echo "" >> tmp/commit.txt
+echo "Changes since $OLD_VERSION:" >> tmp/commit.txt
+# Skip the first three lines of docs/history.md (header, blank line, subheader for this release)
```
Comment (amanda-tarafa) [inline create-release-commit.sh:44]: Maybe add a note about this on the instructions.

**Review by amanda-tarafa — APPROVED** [review]
LGTM

File: RELEASING.md (hunk 1 above)
Comment (jskeet) [reply RELEASING.md:8]: Whoops, thanks. I ended up putting it in the quotes - will remove ", passing"

File: create-release-commit.sh (hunk 2 above)
Comment (jskeet) [reply create-release-commit.sh:34]: Doh! Thanks.

File: create-release-commit.sh (hunk 3 above)
Comment (jskeet) [reply create-release-commit.sh:44]: Yup.

**Pull request merged**